mongoengine/base/metaclasses.py

```diff
@@ -206,7 +209,14 @@ def __new__(mcs, name, bases, attrs):
                         "EmbeddedDocuments (field: %s)" % field.name
                     )
                     raise InvalidDocumentError(msg)
-                f.document_type.register_delete_rule(new_class, field.name, delete_rule)
+                try:
+                    f.document_type.register_delete_rule(
+                        new_class, field.name, delete_rule
```

Comment:
In the ideal world I would cut out `document_type` here and just make fields responsible for implementing `register_delete_rule`.

However, when I tried this I realized that it would break the interface for how to register delete rules and thus any users dynamically setting delete rules would be upset.